plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/binary/KSClassDeclarationDescriptorImpl.kt

```diff
@@ -63,7 +65,7 @@ class KSClassDeclarationDescriptorImpl private constructor(val descriptor: Class
             is FunctionDescriptor -> KSFunctionDeclarationDescriptorImpl.getCached(
                 containingDescriptor
             )
-            else -> throw IllegalStateException()
+            else -> null
```

Comment:
Is this because of containingDescript == null? If so, return in the first line.

Reply:
not necessary, in this case it is containingDesciptor is a package descriptor.